jetty-openid/src/test/java/org/eclipse/jetty/security/openid/JwtEncoder.java

```diff
@@ -0,0 +1,52 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security.openid;
+
+import java.util.Base64;
+
+public class JwtEncoder
+{
+    private static final Base64.Encoder ENCODER = Base64.getUrlEncoder();
+    private static final String DEFAULT_HEADER = "{\"INFO\": \"this is not used or checked in our implementation\"}";
+    private static final String DEFAULT_SIGNATURE = "we do not validate signature as we use the authorization code flow";
+
+    public static String encode(String idToken)
+    {
+        return stripPadding(ENCODER.encodeToString(DEFAULT_HEADER.getBytes())) + "." +
+            stripPadding(ENCODER.encodeToString(idToken.getBytes())) + "." +
+            stripPadding(ENCODER.encodeToString(DEFAULT_SIGNATURE.getBytes()));
+    }
+
+    private static String stripPadding(String paddedBase64)
+    {
+        return paddedBase64.split("=")[0];
+    }
+
+    public static String createIdToken(String provider, String clientId, String subject, String name, long expiry)
+    {
+        return "{" +
```

Comment:
Use StringBuilder?

Also, a comment on the expected format that this method generates might be useful, or a link to a document somewhere that describes it.

Reply:
I just tried this but intellij is suggesting I use string concatenation instead of StringBuilder for better code readability, apparently using StringBuilder in this case is not making the code more efficient.